From 0aef896148a68408d907bf47dc37ed62fbe0b905 Mon Sep 17 00:00:00 2001
From: Vladimir Oltean <olteanv@gmail.com>
Date: Fri, 20 Sep 2019 15:13:59 +0300
Subject: [PATCH] fsl_qbman: Don't compile with -Werror
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc 8 has added a -Wpacked-not-aligned warning, which turns into the
following build errors:

In file included from ./include/linux/fsl_usdpaa.h:17,
                 from drivers/staging/fsl_qbman/dpa_sys.h:67,
                 from drivers/staging/fsl_qbman/dpa_alloc.c:32:
./include/linux/fsl_qman.h:415:1: error: alignment 1 of ‘struct qm_eqcr_entry’ is less than 8 [-Werror=packed-not-aligned]
 } __packed;
 ^
./include/linux/fsl_qman.h:469:3: error: alignment 1 of ‘struct <anonymous>’ is less than 8 [-Werror=packed-not-aligned]
   } __packed ern;
   ^
./include/linux/fsl_qman.h:468:17: error: ‘fd’ offset 15 in ‘struct <anonymous>’ isn’t aligned to 8 [-Werror=packed-not-aligned]
    struct qm_fd fd;
                 ^~
./include/linux/fsl_qman.h:486:3: error: alignment 1 of ‘struct <anonymous>’ is less than 8 [-Werror=packed-not-aligned]
   } __packed dcern;
   ^
./include/linux/fsl_qman.h:485:17: error: ‘fd’ offset 15 in ‘struct <anonymous>’ isn’t aligned to 8 [-Werror=packed-not-aligned]
    struct qm_fd fd;
                 ^~
./include/linux/fsl_qman.h:1679:1: error: alignment 1 of ‘struct qm_mcr_ceetm_cq_peek_pop_xsfdrread’ is less than 8 [-Werror=packed-not-aligned]
 } __packed;
 ^
./include/linux/fsl_qman.h:1677:15: error: ‘fd’ offset 14 in ‘struct qm_mcr_ceetm_cq_peek_pop_xsfdrread’ isn’t aligned to 8 [-Werror=packed-not-aligned]
  struct qm_fd fd;
               ^~

Signed-off-by: Vladimir Oltean <olteanv@gmail.com>
---
 drivers/staging/fsl_qbman/Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/Makefile b/drivers/staging/fsl_qbman/Makefile
index 777d7d34e4f8..37263600f10f 100644
--- a/drivers/staging/fsl_qbman/Makefile
+++ b/drivers/staging/fsl_qbman/Makefile
@@ -1,5 +1,3 @@
-subdir-ccflags-y := -Werror
-
 # Common
 obj-$(CONFIG_FSL_SDK_DPA)		+= dpa_alloc.o
 obj-$(CONFIG_FSL_SDK_DPA)	+= qbman_driver.o
-- 
2.7.4

